warehouse/slave/worker.go

```diff
@@ -636,45 +636,55 @@ func (w *worker) destinationFromSlaveConnectionMap(destinationId, sourceId strin
 	return conn, nil
 }
 
-// HandleSchemaChange checks if the existing column type is compatible with the new column type
-func HandleSchemaChange(existingDataType, currentDataType model.SchemaType, value any) (any, error) {
+// HandleSchemaChange checks if the existing data type (from warehouse schema) is compatible with the inferred data type (from event data)
+func HandleSchemaChange(existingDataType, inferredDataType model.SchemaType, value any) (any, error) {
 	var (
 		newColumnVal any
 		err          error
 	)
 
 	if existingDataType == model.StringDataType || existingDataType == model.TextDataType {
 		// only stringify if the previous type is non-string/text/json
-		if currentDataType != model.StringDataType && currentDataType != model.TextDataType && currentDataType != model.JSONDataType {
+		if inferredDataType != model.StringDataType && inferredDataType != model.TextDataType && inferredDataType != model.JSONDataType {
 			newColumnVal = fmt.Sprintf("%v", value)
 		} else {
 			newColumnVal = value
 		}
-	} else if (currentDataType == model.IntDataType || currentDataType == model.BigIntDataType) && existingDataType == model.FloatDataType {
-		intVal, ok := value.(int)
-		if !ok {
-			err = fmt.Errorf("incompatible schema conversion from %v to %v", existingDataType, currentDataType)
-		} else {
-			newColumnVal = float64(intVal)
+	} else if (inferredDataType == model.IntDataType || inferredDataType == model.BigIntDataType) && existingDataType == model.FloatDataType {
```

Comment:
With the int/float logic inside `HandleSchemaChange`, should we remove it from the other place or do it in the next release?

Reply:
Added a followup task.https://linear.app/rudderstack/issue/WAR-1203/cleanup-int-float-datatype-handling-from-processsinglestagingfile

Even though confident,  would like to make the change in next release.